src/devices/bus/isa/hpblp.h

```diff
@@ -0,0 +1,103 @@
+// license:BSD-3-Clause
+// copyright-holders:Sven Schnelle
+#ifndef MAME_BUS_ISA_HPBLP_H
+#define MAME_BUS_ISA_HPBLP_H
+
+#pragma once
+
+#include "isa.h"
+#include "cpu/m68000/m68000.h"
+#include "machine/tms9914.h"
+#include "bus/ieee488/ieee488.h"
+#include "bus/hp_dio/hp_dio.h"
+#include <queue>
+
+class isa8_hpblp_device :
+		public device_t,
+		public device_isa8_card_interface
+{
+public:
+	isa8_hpblp_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+
+protected:
+	virtual ioport_constructor device_input_ports() const override;
+	virtual void device_start() override;
+	virtual void device_reset() override;
+
+	virtual void device_add_mconfig(machine_config &config) override;
+	virtual const tiny_rom_entry *device_rom_region() const override;
+private:
+	required_device<m68000_device> m_maincpu;
+	required_device<tms9914_device> m_tms9914;
+	required_device<ieee488_device> m_ieee488;
+
+	// ISA part
+	void isamap(address_map &map);
+
+	uint8_t datal_r(offs_t offset);
+	uint8_t datah_r(offs_t offset);
+	uint8_t status_r(offs_t offset);
+	uint8_t addrh_r(offs_t offset);
+	uint8_t addrm_r(offs_t offset);
+	uint8_t addrl_r(offs_t offset);
+	uint8_t reg5_r(offs_t offset);
+	uint8_t reg6_r(offs_t offset);
+
+	void datal_w(offs_t offset, uint8_t data);
+	void datah_w(offs_t offset, uint8_t data);
+	void irq_w(offs_t offset, uint8_t data);
+	void reg2_w(offs_t offset, uint8_t data);
+	void reg3_w(offs_t offset, uint8_t data);
+	void reg4_w(offs_t offset, uint8_t data);
+	void reg5_w(offs_t offset, uint8_t data);
+	void reg6_w(offs_t offset, uint8_t data);
+	void reg7_w(address_space &space, offs_t offset, uint8_t data);
+
+	uint8_t status_val(offs_t offset) const;
+
+	DECLARE_WRITE_LINE_MEMBER(hpblp_interrupt);
+	TIMER_CALLBACK_MEMBER(timer10ms);
+	emu_timer *m_timer_10ms;
+	int m_irq;
```

Comment:
I'm assuming you're talking about these members, as all the other are already explicitly sized types:

bool m_installed;                                                                                                                                                                     
bool m_ack_buscycle;                                                                                                                                                                  
bool m_bus_read;
bool m_bus_write;
bool m_irq_state;
bool m_reset;

So this would mean i need to use uint8_t or similar here? Or maybe i'm just missing some other members?

Reply:
The only one that’s a problem is `m_irq`.  Although C++ `bool` isn’t explicitly sized in the C++ standard, it’s defined as being 1 byte in the Itanium C++ ABI (unlike the C99 Boolean which is 4 bytes on PowerPC and various other targets).  Even if we find an ABI where `bool` isn’t one byte, it can be packed/unpacked in on save/load anyway.

TL;DR just make `m_irq` a `uint8_t` and it’ll be fine.